### PR TITLE
#238 Skip builddirs from components

### LINKED ISSUE #238: KB-H019: it doesn't check builddirs in component
KB-H019 raises an error if it detects cmake files in a folder added to `builddirs` property of a component.

Please see this PR in CCI: https://github.com/conan-io/conan-center-index/pull/2582

### PR DESCRIPTION
**Pull request by uilianries**
fixes #238

/cc @SpaceIm

### CODE REVIEW

File: tests/test_hooks/conan-center/test_cmake_bad_files.py [hunk 1]
```diff
@@ -93,3 +93,21 @@ def test_good_files(self):
...
         output = self.conan(['create', '.', 'name/version@user/channel'])
         self.assertNotIn("ERROR: [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)]", output)
+
+    def test_components(self):
+        conanfile = textwrap.dedent("""\
+                import os
+                from conans import ConanFile, tools
+                class AConan(ConanFile):
+                    def package(self):
+                        tools.save(os.path.join(self.package_folder, {}), "foo")
```
Comment (jgsogo) [inline tests/test_hooks/conan-center/test_cmake_bad_files.py:103]:
```suggestion
                        tools.save(os.path.join(self.package_folder, 'lib', 'cmake', 'FooBar.cmake'), "foo")
```

File: tests/test_hooks/conan-center/test_cmake_bad_files.py [hunk 2]
```diff
@@ -93,3 +93,21 @@ def test_good_files(self):
...
+
+                    def package_info(self):
+                        self.cpp_info.names["cmake_find_package"] = "foobar"
+                        self.cpp_info.components["baz"].names["cmake_find_package"] = "baz"
+                        self.cpp_info.components["baz"].names["cmake_find_package_multi"] = "baz"
+                        self.cpp_info.components["baz"].builddirs = [os.path.join("lib", "cmake")]
+                """)
+        tools.save('conanfile.py', content=conanfile.format('os.path.join("lib", "cmake", "FooBar.cmake")', ""))
```
Comment (jgsogo) [inline tests/test_hooks/conan-center/test_cmake_bad_files.py:111]:
```suggestion
        tools.save('conanfile.py', content=conanfile)
```

**Review by jgsogo — APPROVED** [review]
I think we can simplify a little bit, no need to `conanfile.format`, those are fixed strings.

**Comment by uilianries** [comment]
@jgsogo Indeed, thanks for your suggestions.

**Review by danimtb — APPROVED** [review]

**Pull request merged**